src/bin/cargo/commands/remove.rs

```diff
@@ -0,0 +1,122 @@
+use cargo::core::dependency::DepKind;
+use cargo::ops::cargo_remove::remove;
+use cargo::ops::cargo_remove::RemoveOptions;
+use cargo::ops::resolve_ws;
+use cargo::util::command_prelude::*;
+use cargo::util::toml_mut::manifest::DepTable;
+
+pub fn cli() -> clap::Command<'static> {
+    clap::Command::new("remove")
+        .setting(clap::AppSettings::DeriveDisplayOrder)
+        .about("Remove dependencies from a Cargo.toml manifest file")
+        .args([clap::Arg::new("dependencies")
+            .action(clap::ArgAction::Append)
+            .required(true)
+            .multiple_values(true)
+            .takes_value(true)
+            .value_name("DEP_ID")
+            .help("Dependencies to be removed")])
+        .arg_package("Package to remove from")
+        .arg_manifest_path()
+        .arg_quiet()
+        .arg_dry_run("Don't actually write the manifest")
+        .next_help_heading("SECTION")
+        .args([
+            clap::Arg::new("dev")
+                .long("dev")
+                .conflicts_with("build")
+                .action(clap::ArgAction::SetTrue)
+                .group("section")
+                .help("Remove as development dependency"),
+            clap::Arg::new("build")
+                .long("build")
+                .conflicts_with("dev")
+                .action(clap::ArgAction::SetTrue)
+                .group("section")
+                .help("Remove as build dependency"),
+            clap::Arg::new("target")
+                .long("target")
+                .takes_value(true)
+                .value_name("TARGET")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())
+                .help("Remove as dependency from the given target platform"),
+        ])
+}
+
+pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
+    let dry_run = args.dry_run();
+
+    let workspace = args.workspace(config)?;
+    let packages = args.packages_from_flags()?;
+    let packages = packages.get_packages(&workspace)?;
+    let spec = match packages.len() {
+        0 => {
+            return Err(CliError::new(
+                anyhow::format_err!("no packages selected. Please specify one with `-p <PKG_ID>`"),
+                101,
```

Comment:
side note: A future improvement here is to display the list of available packages, like you see if you run `cargo check -p`